### PR TITLE
Private chat messages showing in correct order for both participants

### PR DESCRIPTION
**Pull request by IvanBuljovcic**
* When a user sends a message, only the recipient saw the message, not the sender.
* All messages were in random order, now sorted by timestamp **ts_utc** in reverse (last message first)

### CODE REVIEW

File: src/components/molecules/ChatList/ChatList.tsx
Comment (0xdevalias) [inline src/components/molecules/ChatList/ChatList.tsx:94]: Probably want to useMemo this

File: src/components/molecules/ChatList/ChatList.tsx
Comment (0xdevalias) [inline src/components/molecules/ChatList/ChatList.tsx:112]: Extract this from the JSX + `useMemo`. If it's the only place `sortedMessages` is needed/used, then you could probably combine this + my above comment within the same `useMemo` block.

File: src/components/molecules/ChatList/ChatList.tsx
Comment (0xdevalias) [inline src/components/molecules/ChatList/ChatList.tsx:103]: Is there something more unique/good that we can use here in place of index? Maybe the `ts_utc` + who sent it or similar if messages don't have an actual UID?

File: src/components/molecules/ChatList/ChatMessage.tsx
Comment (0xdevalias) [inline src/components/molecules/ChatList/ChatMessage.tsx:56]: [nit] Since we check `!user` above, do we need to null check `user?.uid` here?
```suggestion
  const isMe = message.from === user.uid;
```

File: src/components/molecules/ChatList/ChatMessage.tsx
Comment (0xdevalias) [inline src/components/molecules/ChatList/ChatMessage.tsx:104]: This will cause a type error once https://github.com/sparkletown/sparkle/pull/897 gets merged.
```suggestion
                {formatUtcSeconds(message.ts_utc.seconds)}
```

**Review by 0xdevalias — CHANGES_REQUESTED** [review]
Some `useMemo` related changes + minor comments. Once those are addressed, re-request review, but if I am not around, feel free to dismiss my stale review + get @denisdimitrov or similar to give the approval stamp.

File: src/components/molecules/ChatList/ChatList.tsx
Comment (denisdimitrov) [inline src/components/molecules/ChatList/ChatList.tsx:99]: If someone sends 2 messages in less than a second, does that mean that there will be duplicated keys?

Comment (0xdevalias) [reply src/components/molecules/ChatList/ChatList.tsx:99]: I believe it will. Do we have milliseconds available on that object?

File: src/components/molecules/ChatList/ChatList.tsx
Comment (0xdevalias) [reply src/components/molecules/ChatList/ChatList.tsx]: resolved in https://github.com/sparkletown/sparkle/pull/921/commits/f69bd1a816115991a5ca4257830f370237e55697

File: src/components/molecules/ChatList/ChatList.tsx
Comment (0xdevalias) [reply src/components/molecules/ChatList/ChatList.tsx]: resolved in https://github.com/sparkletown/sparkle/pull/921/commits/f69bd1a816115991a5ca4257830f370237e55697

File: src/components/molecules/ChatList/ChatList.tsx
Comment (0xdevalias) [reply src/components/molecules/ChatList/ChatList.tsx]: resolved in https://github.com/sparkletown/sparkle/pull/921/commits/f69bd1a816115991a5ca4257830f370237e55697 though see also https://github.com/sparkletown/sparkle/pull/921#discussion_r537551003

File: src/components/molecules/ChatList/ChatMessage.tsx
Comment (0xdevalias) [reply src/components/molecules/ChatList/ChatMessage.tsx]: https://github.com/sparkletown/sparkle/pull/921/commits/f69bd1a816115991a5ca4257830f370237e55697

File: src/components/molecules/ChatList/ChatMessage.tsx
Comment (0xdevalias) [reply src/components/molecules/ChatList/ChatMessage.tsx]: resolved in https://github.com/sparkletown/sparkle/pull/921/commits/f69bd1a816115991a5ca4257830f370237e55697

**Review by 0xdevalias — COMMENTED** [review]
Thanks for addressing the issues in my last review.

I'll hold off on approving this as due to @denisdimitrov 's comment in https://github.com/sparkletown/sparkle/pull/921#pullrequestreview-546202951, but happy for him to review/approve once he's happy there.

**Review by thecadams — APPROVED** [review]

**Comment by thecadams** [comment]
@0xdevalias I'm going to consider your review as having been completed as I've addressed the concern about duplicate keys by adding a reference to `nanoseconds`. Going to dismiss your changes request and merge. Thanks for the review.

**Pull request merged**
